File: FrontEnd/lib/domain/interfaces/usecases/reminders/i_mark_reminder_completed_use_case.dart
/// @file i_mark_reminder_completed_use_case.dart
/// @description Interfaz del caso de uso para marcar un recordatorio como completado.
/// @module Reminders
/// @layer Domain
library;
abstract interface class IMarkReminderCompletedUseCase {
  /// Marca el recordatorio [reminderId] como completado.
  Future<void> execute(String reminderId);
}

File: BackEnd/src/domain/interfaces/usecases/reminders/IMarkReminderCompletedUseCase.ts
/**
 * @file IMarkReminderCompletedUseCase.ts
 * @description Interfaz del caso de uso para marcar un recordatorio como completado.
 * @module Reminders
 * @layer Domain
 */
export interface IMarkReminderCompletedUseCase {
  execute(reminderId: string, userId: string): Promise<void>;
}
